src/components/Profile.js

```diff
@@ -57,35 +54,26 @@ class Profile extends Component<Props> {
       && !this.props.trustedByUids.has(this.props.authMemberDoc.id);
   }
 
-  isOwnProfile(uid) {
-    return this.props.authMemberDoc && this.props.authMemberDoc.uid === uid;
+  isOwnProfile() {
+    const { authFirebaseUser, memberDoc } = this.props;
+    return authFirebaseUser && memberDoc && authFirebaseUser.uid === memberDoc.id;
   }
 
-
   render() {
-    const { authFirebaseUser, authIsLoaded, isMyProfile, memberDoc } = this.props;
-    if (!authIsLoaded) {
-      return <Loading />;
-    }
-    if (isMyProfile && !authFirebaseUser) {
-      return (
-        <div>
-          <span>You need to </span>
-          <Redirect to="/login" />
-        </div>
-      );
-    }
-    if (!memberDoc) {
+    const { authFirebaseUser, authIsLoaded, memberDoc } = this.props;
+    if (!authIsLoaded || !memberDoc) {
       return <Loading />;
     }
+    const ownProfile = this.isOwnProfile();
     if (!memberDoc || !memberDoc.get('mid')) {
       if (!this.props.memberId) {
-        // TODO make into it's own <ProfilePending /> component
-        return <FormattedMessage
-          id="invite_pending"
+        return <FormattedMessage  // TODO this does not work
```

Comment:
What about this doesn't work? / is there a reason you're not using the new `InviteMissing` component?